gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -898,49 +858,31 @@ private Path getHdfsLogDir(Path appWorkDir) throws IOException {
     return logRootDir;
   }
 
-  private AbstractYarnAppSecurityManager buildSecurityManager() throws IOException {
+  /**
+   *
+   * @return
+   * @throws IOException
+   */
+  private AbstractAppSecurityManager buildSecurityManager() throws IOException {
     Path tokenFilePath = new Path(this.fs.getHomeDirectory(), this.applicationName + Path.SEPARATOR +
         GobblinYarnConfigurationKeys.TOKEN_FILE_NAME);
+    String securityManagerClassName = ConfigUtils.getString(config, GobblinYarnConfigurationKeys.SECURITY_MANAGER_CLASS, GobblinYarnConfigurationKeys.DEFAULT_SECURITY_MANAGER_CLASS);
 
-    ClassAliasResolver<AbstractYarnAppSecurityManager> aliasResolver = new ClassAliasResolver<>(
-        AbstractYarnAppSecurityManager.class);
     try {
-     return (AbstractYarnAppSecurityManager) GobblinConstructorUtils.invokeLongestConstructor(Class.forName(aliasResolver.resolve(
-          ConfigUtils.getString(config, GobblinYarnConfigurationKeys.SECURITY_MANAGER_CLASS, GobblinYarnConfigurationKeys.DEFAULT_SECURITY_MANAGER_CLASS))), this.config, this.helixManager, this.fs,
-          tokenFilePath);
+      if (helixClusterLifecycleManager.isPresent()) {
+        HelixManager helixManager = helixClusterLifecycleManager.get().getHelixManager();
+        ClassAliasResolver<AbstractYarnAppSecurityManager> aliasResolver = new ClassAliasResolver<>(AbstractYarnAppSecurityManager.class);
+        return (AbstractYarnAppSecurityManager) GobblinConstructorUtils.invokeLongestConstructor(Class.forName(aliasResolver.resolve(securityManagerClassName)), this.config, helixManager, this.fs,
+            tokenFilePath);
+      }
+
+      ClassAliasResolver aliasResolver = new ClassAliasResolver<>(AbstractAppSecurityManager.class);
+      return (AbstractAppSecurityManager) GobblinConstructorUtils.invokeLongestConstructor(Class.forName(aliasResolver.resolve(securityManagerClassName)), this.config, this.fs, tokenFilePath);
```

Comment:
somewhat confusing to have alt. `ClassAliasResolver`s, one typed and the other `CAR<?>`.

couldn't we just do one and take advantage of the ability to pass `null` to the second param of the ctor by doing:
```
helixClusterLifecycleManager.map(HCLM::getHelixManager).orElse(null)
```

Reply:
aah, ok... never hit that so never would have suspected.  just add a code comment to convey this critical tidbit